src/services/CertificateGenerationService.ts

```diff
@@ -507,9 +541,49 @@ class CertificateGenerationService {
                 };
                 console.log("CHECK HERE DOCGENPAYLOAD -> ", docGenPayloadAdr);
                 return docGenPayloadAdr;
+            case CERTIFICATE_DATA.IVA_DATA:
+                const ivaFailDetailsForDocGen = {
+                    vin: testResult.vin,
+                    serialNumber: testResult.vehicleType === "trl" ? testResult.trailerId : testResult.vrm,
+                    vehicleTrailerNrNo: testResult.vehicleType === "trl" ? testResult.trailerId : testResult.vrm,
+                    testCategoryClass: testResult.euVehicleCategory,
+                    testCategoryBasicNormal: this.isBasicIvaTest(testResult.testTypes.testTypeId) ? IVA_30.BASIC : IVA_30.NORMAL,
+                    make: testResult.make,
+                    model: testResult.model,
+                    bodyType: testResult.bodyType?.description,
+                    date: moment(testResult.testTypes.testTypeStartTimestamp).format("DD/MM/YYYY"),
+                    testerName: testResult.testerName,
+                    reapplicationDate: moment(testResult.testTypes.testTypeStartTimestamp)
+                        .add(6, "months")
+                        .subtract(1, "day")
+                        .format("DD/MM/YYYY"),
+                    station: testResult.testStationName,
+                    additionalDefects:
+                        testResult.testTypes.customDefects && testResult.testTypes.customDefects.length > 0
+                            ? testResult.testTypes.customDefects
+                            : [{ defectName: IVA_30.EMPTY_CUSTOM_DEFECTS, defectNotes: ""}],
+                    requiredStandards: testResult.testTypes.requiredStandards
+                };
+                return ivaFailDetailsForDocGen;
         }
     }
 
+    /**
+     * Returns a boolean value indicating whether the test type is a basic IVA test
+     * @param testTypeId - the test type ID on the test result
+     */
+    public isBasicIvaTest = (testTypeId: string): boolean => {
+        const basicIvaTests: string[] = [
+            "125",
+            "129",
+            "154",
+            "158",
+            "159",
+            "185"
+        ];
+        return basicIvaTests.includes(testTypeId);
+    }
```

Comment:
you could also probably streamline this something like this. A similar thing could be done for the isaIvaTest method
```suggestion
   public isBasicIvaTest = (testTypeId): boolean =>
    ["125", "129", "154", "158", "159", "185"].includes(testTypeId);
```